python/lsst/analysis/tools/atools/diaFakeMetrics.py

```diff
@@ -170,6 +148,108 @@ def finalize(self):
+
+    snrMax = Field[float](doc="Maximum SNR for fake sources metric calculation.", default=np.inf)
+
+    fluxType = Field[str](
+        "Flux type for fake sources metric calculation.", default="forced_base_PsfFlux_instFlux"
+    )
+
+    fakeFlagsWhenTrue = ListField[str](
+        "Flags for fake source cleaning before metrics calculation.", default=[]
+    )
+
+    fakeFlagsWhenFalse = ListField[str](
+        "Flags for fake source cleaning before metrics calculation.",
+        default=[
+            "forced_base_PixelFlags_flag_bad",
+            "forced_base_LocalBackground_flag",
+            "forced_base_PixelFlags_flag_interpolated",
+            "forced_base_PixelFlags_flag_edgeCenter",
+        ],
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+
+    def finalize(self):
+        # There is no need to calculate the SNR as it is already estimated
+        # Select the fake sources using their SNR values for the given range
+        # and flux type estimation
+
+        self.process.filterActions.fakeSourcesAssocDiaSrcId = MultiCriteriaDownselectVector(
+            vectorKey="isAssocDiaSource"
+        )
+
+        self.process.filterActions.fakeSourcesAssocDiaSrcId.selectors.snrange = RangeSelector(
+            vectorKey=f"{self.fluxType}_SNR", maximum=self.snrMax, minimum=self.snrMin
+        )
+
+        self.process.filterActions.fakeSourcesAssocDiaSrcId.selectors.fakeFlags = FlagSelector(
+            selectWhenFalse=self.fakeFlagsWhenFalse, selectWhenTrue=self.fakeFlagsWhenTrue
+        )
+
+        self.process.calculateActions.numTotalFakeAssocSources = CountAction(
+            vectorKey="fakeSourcesAssocDiaSrcId"
+        )
+
+        self.process.calculateActions.numFoundFakeAssocSources = CountAction(
+            vectorKey="fakeSourcesAssocDiaSrcId", op="gt", threshold=0
+        )
+
+        self.process.calculateActions.fractionFoundFakesAssocDiaAll = FracThreshold(
+            vectorKey="fakeSourcesAssocDiaSrcId", op="gt", threshold=0
+        )
+
+        # the units for the quantity (count, an astropy quantity)
+        self.produce.metric.units = {
+            "numTotalFakeAssocSources": "ct",
+            "numFoundFakeAssocSources": "ct",
+            "fractionFoundFakesAssocDiaAll": "",
+        }
+
+
+class FractionFoundFakesAssocDiaMagMetric(AnalysisTool):
+    """Calculate the fraction of fake sources found in AssocDiasrcs within
+    the given magnitude range"""
+
+    parameterizedBand: bool = False
+
+    magMin = Field[float](doc="Minimum magnitude for fake sources metric calculation.", default=18)
+
+    magMax = Field[float](doc="Maximum magnitude for fake sources metric calculation.", default=22)
+
+    fakeFlagsWhenTrue = ListField[str](
+        "Flags for fake source cleaning before metrics calculation.", default=[]
+    )
+
+    fakeFlagsWhenFalse = ListField[str](
+        "Flags for fake source cleaning before metrics calculation.",
```

Comment:
Please reword these two docstrings to clarify their use, i.e. how each is used for cleaning.